store/store.go

```diff
@@ -773,31 +765,16 @@ func (s *Store) decoratePurchases(snaps []*snap.Info, channel string, user *auth
 	}
 
 	var err error
-	var purchasesURL *url.URL
-
-	if len(snaps) == 1 {
-		// If we only have a single snap, we should only find the purchases for that snap
-		purchasesURL, err = s.purchasesURI.Parse(snaps[0].SnapID + "/")
-		if err != nil {
-			return err
-		}
-		q := purchasesURL.Query()
-		q.Set("include_item_purchases", "true")
-		purchasesURL.RawQuery = q.Encode()
-	} else {
-		// Inconsistently, global search implies include_item_purchases.
-		purchasesURL = s.purchasesURI
-	}
 
-	purchases, err := s.getPurchasesFromURL(purchasesURL, channel, user)
+	purchases, err := s.getPurchasesFromURL(s.purchasesURI, channel, user)
 	if err != nil {
 		return err
 	}
 
 	// Group purchases by snap ID.
-	purchasesByID := make(map[string][]*purchase)
-	for _, purchase := range purchases {
-		purchasesByID[purchase.SnapID] = append(purchasesByID[purchase.SnapID], purchase)
+	purchasesByID := make(map[string]bool)
```

Comment:
John: After discussion, we resolved this misunderstanding, and agreed on naming for the map.